content/blog/using-the-hpegl-hpe-greenlake-and-morpheus-terraform-providers-in-concert-to-create-provision-and-manage-hpe-pce-private-cloud-enterprise-vmaas-instances.md

```diff
@@ -0,0 +1,1492 @@
+module.instance_cloud_2.hpegl_vmaas_instance.sample_vm[0]: Still creating... [3m40s elapsed]
+module.instance_cloud_1.hpegl_vmaas_instance.sample_vm[0]: Still creating... [3m40s elapsed]
+module.instance_cloud_2.hpegl_vmaas_instance.sample_vm[0]: Still creating... [3m50s elapsed]
+module.instance_cloud_1.hpegl_vmaas_instance.sample_vm[0]: Still creating... [3m50s elapsed]
+module.instance_cloud_1.hpegl_vmaas_instance.sample_vm[0]: Still creating... [4m0s elapsed]
+module.instance_cloud_2.hpegl_vmaas_instance.sample_vm[0]: Still creating... [4m0s elapsed]
+module.instance_cloud_2.hpegl_vmaas_instance.sample_vm[0]: Still creating... [4m10s elapsed]
+module.instance_cloud_1.hpegl_vmaas_instance.sample_vm[0]: Still creating... [4m10s elapsed]
+module.instance_cloud_1.hpegl_vmaas_instance.sample_vm[0]: Creation complete after 4m20s [id=637]
+module.instance_cloud_2.hpegl_vmaas_instance.sample_vm[0]: Still creating... [4m20s elapsed]
+module.instance_cloud_2.hpegl_vmaas_instance.sample_vm[0]: Creation complete after 4m21s [id=634]
+
+Apply complete! Resources: 14 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+instance_ids_cloud_1 = "637"
+instance_ids_cloud_2 = "634"
+instance_layout_id_cloud_1 = "3562"
+instance_layout_id_cloud_2 = "3559"
+instance_layout_name_cloud_1 = "cfe_todo_app_frontend--40272"
+instance_layout_name_cloud_2 = "cfe_todo_app_frontend--1661"
+instance_names_cloud_1 = "CFE morpheus-vmaas sample VM 1-36444-0"
+instance_names_cloud_2 = "CFE morpheus-vmaas sample VM 1-10027-0"
+node_type_id_cloud_1 = "5113"
+node_type_id_cloud_2 = "5110"
+node_type_name_cloud_1 = "cfe_tf_example_node_type--40272"
+node_type_name_cloud_2 = "cfe_tf_example_node_type--1661"
+```
+
+# Summary 
+
+When using Terraform (or OpenTofu) to interact with HPE PCE VMaaS there are two complementary providers: 
+
+* hpegl (HPE GreenLake) 
+* Morpheus 
+
+ When used in concert these two providers enable a rich set of IaC functionality.  The hpegl provider has a *Data Source* named *hpegl_vmaas_morpheus_details* which can be used to retrieve an access token and URL for the on-premise Morpheus instance associated with the specific VMaaS *location*.  These can be used as inputs to the provider stanza for the Morpheus provider. 
+
+We have included example HCL (Hashicorp Configuration Language) for two separate VMaaS configurations: 
+
+* One Location, One VMaaS Cloud with One VMaaS Instance 
+* One Location, Two VMaaS Clouds each with One VMaaS Instance 
+
+ The HCL for the second configuration is built around a pair of HCL modules that can be combined in various ways to work with a number of different VMaaS configurations, including those with more than one on-premise Morpheus instance. 
+
+We encourage PCE customers to use these example HCL bundles as a starting point for their own investigations into using the two providers (hpegl and Morpheus) for IaC with their environments.
```

Comment:
We encourage PCE customers to use these example HCL bundles as a starting point for their own investigations into using the two providers (hpegl and Morpheus) for IaC with their environments.

Suggestion:
HPE GreenLake for Private Cloud Enterprise are encouraged to use these example HCL bundles....

Reply:
Done.